dashproducts.php

```diff
@@ -221,7 +221,12 @@ public function getTableBestSellers($date_from, $date_to)
             if (!Validate::isLoadedObject($product_obj)) {
                 continue;
             }
-            $category = new Category($product_obj->getDefaultCategory()['id_category_default'], $this->context->language->id);
+
+            $productCategoryId = $product_obj->getDefaultCategory();
+            if (is_array($product_obj->getDefaultCategory()) && isset($product_obj->getDefaultCategory()['id_category_default'])) {
+                $productCategoryId = $product_obj->getDefaultCategory()['id_category_default'];
+            }
+            $category = new Category($productCategoryId, $this->context->language->id);
```

Comment:
```suggestion
            $productCategoryId = $product_obj->getDefaultCategory();
            if (is_array($productCategoryId) && isset($productCategoryId['id_category_default'])) {
                $productCategoryId = $productCategoryId['id_category_default'];
            }
            $category = new Category($productCategoryId, $this->context->language->id);
```
Saves 3 database queries